espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/handlers/Uiautomator.java

```diff
@@ -0,0 +1,50 @@
+package io.appium.espressoserver.lib.handlers;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.List;
+
+import androidx.test.uiautomator.By;
+import androidx.test.uiautomator.BySelector;
+import androidx.test.uiautomator.UiObject2;
+import io.appium.espressoserver.lib.handlers.exceptions.AppiumException;
+import io.appium.espressoserver.lib.helpers.ReflectionUtils;
+import io.appium.espressoserver.lib.model.UiautomatorParams;
+
+import static io.appium.espressoserver.lib.helpers.AndroidLogger.logger;
+import static io.appium.espressoserver.lib.helpers.InteractionHelper.getUiDevice;
+
+public class Uiautomator implements RequestHandler<UiautomatorParams, Object> {
+
+    @Override
+    public Object handle(UiautomatorParams params) throws AppiumException {
+        logger.info("Invoking Uiautomator Methods");
+
+        ArrayList<Object> result = new ArrayList<>();
+        String byMethodName = params.getBy();
+        String value = params.getValue();
+        String action = params.getAction();
```

Comment:
should we also support action arguments?

Reply:
I personally do not feel there is a need for methods with arguments. Purpose of this code is to support common operations like click, longClick, getText,  getContentDescription etc which do not take any arguments. Other actions like setText can be workaround by click() and `adb shell input text`. In case we need to support more complex actions then this approach will never be sufficient

Reply:
it is anyway just a proposal. I agree we should keep it simple to the last extent possible